dll/appcompat/shims/layer/aero.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * PROJECT:     ReactOS 'Layers' Shim library
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Display settings related shims
+ * COPYRIGHT:   Copyright 2024 William Kent (wjk011@gmail.com)
+ */
+
+#include "string.h"
+#define WIN32_NO_STATUS
+#include <windef.h>
+#include <winbase.h>
+#include <shimlib.h>
+
+#define SHIM_NS             AeroThemeName
+#include <setup_shim.inl>
+
+HRESULT WINAPI SHIM_OBJ_NAME(GetCurrentThemeNameRedirect)(
+    LPWSTR pszThemeName, int cchThemeName, LPWSTR pszColor, int cchColor, LPWSTR pszSize, int cchSize
+) {
+    wcsncpy(pszThemeName, L"Aero", cchThemeName);
+    wcsncpy(pszColor, L"NormalColor", cchColor);
+    wcsncpy(pszSize, L"Normal", cchSize);
```

Comment:
Shouldn't you use the safe-string helper functions in order to ensure that the strings will always be NUL-terminated?

In addition, according to [GetCurrentThemeName function (uxtheme.h)](https://learn.microsoft.com/en-us/windows/win32/api/uxtheme/nf-uxtheme-getcurrentthemename), the last two parameters `pszColor` and `pszSize` can be set to NULL by the caller, so you must check for that before copying over the data.